File: server/src/main/java/edu/setokk/astrocluster/response/GetAnalysisResponse.java
package edu.setokk.astrocluster.response;

import edu.setokk.astrocluster.model.dto.AnalysisDto;
import edu.setokk.astrocluster.model.dto.PercentagePerClusterDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.Singular;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public final class GetAnalysisResponse {
    private AnalysisDto analysisData;
    @Singular("addPercentagePerCluster") private List<PercentagePerClusterDto> percentagesPerCluster;
}

File: client/src/app/core/response/GetAnalysisResponse.ts
import {AnalysisDto} from "../model/AnalysisDto";
import {PercentagePerClusterDto} from "../model/PercentagePerClusterDto";

export interface GetAnalysisResponse {
    analysisData: AnalysisDto;
    percentagesPerCluster: PercentagePerClusterDto[];
}